subsys/mpsl/fem/common/mpsl_fem_utils.c

```diff
@@ -16,6 +16,10 @@
 #include <nrfx_dppi.h>
 #endif
 
+#ifdef DPPI_PRESENT
+static nrfx_dppi_t dppi = NRFX_DPPI_INSTANCE(0);
+#endif
```

Comment:
The `dppi` variable is used only as a kind of a "handle" to given resources represented by `nrfx_dppi_t`.
See https://github.com/e-rk/hal_nordic/blob/d7317a9cd28a845dfd174e14c16ba3def1c02a78/nrfx/drivers/src/nrfx_dppi.c#L527
The `nrfx_dppi_channel_alloc` has `const` qualifier for the `p_instance`, so the `dppi` variable itself (not referring to resource it "points to") will not change. 
```c
nrfx_err_t nrfx_dppi_channel_alloc(nrfx_dppi_t const * p_instance, uint8_t * p_channel)
```
This is true for all other "nrfx_dppi_xxxxxx" calls.

Could the variable be moved to be inside appropriate function as a local variable?
In this case into `mpsl_fem_utils_ppi_channel_alloc` so that it is a local non-static variable (on stack).

This could be done for all places you introduce the `dppi` variable .

Justification:
- The `static nrfx_dppi_t dppi` will exist for application lifetime, but the `dppi` object itself does not change.
- The initializer `NRFX_DPPI_INSTANCE` is prepared by preprocessor.
- There are multiple such variables, each takes space.
- There is only one underlying allocator per given DPPIC regardless of how many `nrfx_dppi_t dppi` we have used.
- They are used rather in non time-critical code, there is no need to have them precalculated.

Reply:
Moved instance to local scope, with one exception.